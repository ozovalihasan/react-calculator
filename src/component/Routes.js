import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import App from './App';
import Home from './Home';
import Navbar from './Navbar';
import Quote from './Quote';

const Routes = () => (
  <BrowserRouter>
    <Navbar />
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/calculator" component={App} />
      <Route exact path="/quote" component={Quote} />
    </Switch>
  </BrowserRouter>
);

export default Routes;
