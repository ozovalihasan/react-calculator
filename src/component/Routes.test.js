import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Routes from './Routes';

describe('<Routes />', () => {
  it('shows correct pages when any link is clicked', () => {
    render(<Routes />);
    fireEvent.click(screen.queryByText('Home'));
    expect(screen.queryByText(/Welcome to our page/)).toBeTruthy();
    expect(screen.queryByText(/Let's do some math/)).toBeFalsy();
    expect(screen.queryByText(/Mathematics is not about numbers/)).toBeFalsy();

    fireEvent.click(screen.queryByText('Calculator'));
    expect(screen.queryByText(/Welcome to our page/)).toBeFalsy();
    expect(screen.queryByText(/Let's do some math/)).toBeTruthy();
    expect(screen.queryByText(/Mathematics is not about numbers/)).toBeFalsy();

    fireEvent.click(screen.queryByText('Quote'));
    expect(screen.queryByText(/Welcome to our page/)).toBeFalsy();
    expect(screen.queryByText(/Let's do some math/)).toBeFalsy();
    expect(screen.queryByText(/Mathematics is not about numbers/)).toBeTruthy();
  });

  it('renders correctly', () => {
    const component = renderer.create(<Routes />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
