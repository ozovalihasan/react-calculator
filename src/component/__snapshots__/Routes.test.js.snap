// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> renders correctly 1`] = `
Array [
  <div
    className="navbar main"
  >
    <ul
      className="navbar container"
    >
      <h2
        className="navbar title"
      >
        Math Magicians
      </h2>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      |
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      |
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="quote main"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul Thurston
  </div>,
]
`;
